common/app/conf/HealthCheck.scala

```diff
@@ -0,0 +1,153 @@
+package conf
+
+import common._
+import org.joda.time.DateTime
+import play.api.{Mode, Play}
+import play.api.libs.ws.WS
+import play.api.mvc._
+import play.api.{Application => PlayApp, GlobalSettings}
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+/*
+* Healthcheck endpoints are requested periodically
+* Results are stored in a cache
+* Goal: Requests made to the healthcheck endpoint by the ELBs are not 'inlined'
+* */
+
+private[conf] case class HealthCheckResult(url: String,
+                                           result: Either[Throwable, Int], // Status code if success, throwable otherwise
+                                           date: DateTime = DateTime.now,
+                                           expiration: Duration = 10.seconds) {
+  private def age = Duration(DateTime.now.getMillis - date.getMillis, MILLISECONDS)
+  private def expired = age > expiration
+  def recentlySucceed = result.fold(_ => false, _ == 200) && !expired
+  def formattedResult = result match {
+    case Left(t) => s"Error: ${t.getLocalizedMessage}"
+    case Right(status) => status
+  }
+  def formattedDate = if(expired) s"${date} (Expired)" else date
+}
+
+object HealthCheckResult {
+  def apply(url: String, status: Int) = {
+    new HealthCheckResult(url, Right(status))
+  }
+  def apply(url: String, error: Throwable) = {
+    new HealthCheckResult(url, Left(error))
+  }
+  def apply(url: String, status: Int, date: DateTime) = {
+    new HealthCheckResult(url, Right(status), date)
+  }
+  def apply(url: String, error: Throwable, date: DateTime) = {
+    new HealthCheckResult(url, Left(error), date)
+  }
+  def apply(url: String, status: Int, date: DateTime, expiration: Duration) = {
+    new HealthCheckResult(url, Right(status), date, expiration)
+  }
+  def apply(url: String, error: Throwable, date: DateTime, expiration: Duration) = {
+    new HealthCheckResult(url, Left(error), date, expiration)
+  }
+}
```

Comment:
You are right. I wanted to but I ended up only using this in the tests. Will move